tt_metal/detail/tt_metal.hpp

```diff
@@ -284,9 +284,7 @@ namespace tt::tt_metal{
             static std::mutex cq_creation_mutex;
             {
                 std::lock_guard<std::mutex> lock(cq_creation_mutex);
-                if (not command_queues[id] or (command_queues[id] and command_queues[id]->device != device)) {
-                    command_queues[device->id()] = std::make_unique<CommandQueue>(device);
-                }
+                command_queues[device->id()] = std::make_unique<CommandQueue>(device, id);
```

Comment:
could you add a comment to the GetCommandQueue function explaining why a new CQ instance is created each time this API is called

Reply:
isn't this a bug?

Reply:
@TT-billteng  This is not a bug. Creating a command queue now is very light-weight (aka kernels not sent to device). I wanted to ensure that we don't ever get a case where the command queue is not defined for the device. I can add a comment.